File: server_j/src/main/java/com/rd/zngp/model/Record.java
package com.rd.zngp.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.LocalDateTime;

/**
 * Record model, mirroring server/internal/model/record.go
 */
public class Record {

    @JsonProperty("id")
    public String id;

    @JsonProperty("title")
    public String title;

    @JsonProperty("description")
    public String description;

    @JsonProperty("inspector_name")
    public String inspectorName;

    @JsonProperty("customer_name")
    public String customerName;

    @JsonProperty("customer_address")
    public String customerAddress;

    @JsonProperty("inspection_date")
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    public LocalDateTime inspectionDate;

    @JsonProperty("source_type")
    public String sourceType = "RECORDING";

    @JsonProperty("audio_file_path")
    public String audioFilePath;

    @JsonProperty("audio_duration")
    public double audioDuration;

    @JsonProperty("transcript_text")
    public String transcriptText;

    @JsonProperty("transcript_status")
    public String transcriptStatus = "PENDING";

    @JsonProperty("inspection_status")
    public String inspectionStatus = "NONE";

    @JsonProperty("created_at")
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    public LocalDateTime createdAt;

    @JsonProperty("updated_at")
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    public LocalDateTime updatedAt;

    /**
     * RecordCreateRequest, mirroring server internal/model RecordCreateRequest.
     */
    public static class CreateRequest {
        @JsonProperty("id")
        public String id;

        @JsonProperty("title")
        public String title;

        @JsonProperty("description")
        public String description;

        @JsonProperty("inspector_name")
        public String inspectorName;

        @JsonProperty("customer_name")
        public String customerName;

        @JsonProperty("customer_address")
        public String customerAddress;

        @JsonProperty("inspection_date")
        public String inspectionDate;

        @JsonProperty("source_type")
        public String sourceType;
    }
}

File: srv_g/internal/model/record.go
package model

import (
	"time"
)

// Record represents a voice recording uploaded from the app
type Record struct {
	ID                string    `json:"id" gorm:"primaryKey"`
	Title             string    `json:"title"`
	Description       string    `json:"description"`
	InspectorName     string    `json:"inspector_name"`
	CustomerName      string    `json:"customer_name"`
	CustomerAddress   string    `json:"customer_address"`
	InspectionDate    time.Time `json:"inspection_date"`
	SourceType        string    `json:"source_type" gorm:"default:RECORDING"`
	AudioFilePath     string    `json:"audio_file_path"`
	AudioDuration     float64   `json:"audio_duration"`
	TranscriptText    string    `json:"transcript_text"`
	TranscriptStatus  string    `json:"transcript_status" gorm:"default:PENDING"`
	InspectionStatus  string    `json:"inspection_status" gorm:"default:NONE"`
	CreatedAt         time.Time `json:"created_at"`
	UpdatedAt         time.Time `json:"updated_at"`
}

func (Record) TableName() string { return "records" }

// RecordCreateRequest is the JSON part of the upload request
type RecordCreateRequest struct {
	ID              string `json:"id"`               // optional, App can provide UUID
	Title           string `json:"title"`
	Description     string `json:"description"`
	InspectorName   string `json:"inspector_name"`
	CustomerName    string `json:"customer_name"`
	CustomerAddress string `json:"customer_address"`
	InspectionDate  string `json:"inspection_date"`  // RFC3339 or "2006-01-02T15:04"
	SourceType      string `json:"source_type"`      // RECORDING / IMPORT
}